src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -294,4 +348,54 @@ public static ObjectPool<ExpressionVariableFinder> CreatePool()
         }
         #endregion
     }
+
+    internal class ExpressionFieldFinder : ExpressionVariableFinder<Symbol>
+    {
+        private SourceMemberContainerTypeSymbol _containingType;
+        private DeclarationModifiers _modifiers;
+        private FieldSymbol _containingFieldOpt;
+
+        internal static void FindExpressionVariables(
+            ArrayBuilder<Symbol> builder,
+            CSharpSyntaxNode node,
+            SourceMemberContainerTypeSymbol containingType,
+            DeclarationModifiers modifiers,
+            FieldSymbol containingFieldOpt)
+        {
+            if (node == null)
+            {
+                return;
+            }
+
+            var finder = s_poolInstance.Allocate();
+            finder._containingType = containingType;
+            finder._modifiers = modifiers;
+            finder._containingFieldOpt = containingFieldOpt;
+
+            finder.FindExpressionVariables(builder, node);
+
+            finder._containingType = null;
+            finder._modifiers = DeclarationModifiers.None;
+            finder._containingFieldOpt = null;
+            s_poolInstance.Free(finder);
+        }
+
+        protected override Symbol MakePatternVariable(DeclarationPatternSyntax node, SyntaxNode nodeToBind)
+        {
+            throw ExceptionUtilities.Unreachable;
```

Comment:
That is Ok, a work item will be opened to add support for patterns in Script.

---
In reply to: [79079998](https://github.com/dotnet/roslyn/pull/13818#discussion_r79079998) [](ancestors = 79079998,79079667)